File: modulo-01-fundamentos-arquitectura/codigo/src/main/java/pe/joedayz/microservicios/modulo01/ddd/order/OrderStatus.java
package pe.joedayz.microservicios.modulo01.ddd.order;

/**
 * Ciclo de vida del pedido. El agregado {@link Order} controla las transiciones
 * validas entre estos estados (es un invariante: no se puede pasar de CONFIRMED a DRAFT).
 *
 * <pre>
 *   DRAFT ──place()──► PLACED ──confirm()──► CONFIRMED
 *                        │
 *                        └──cancel()──► CANCELLED
 * </pre>
 */
public enum OrderStatus {
    DRAFT,
    PLACED,
    CONFIRMED,
    CANCELLED
}

File: modulo-01-fundamentos-arquitectura/codigo-dotnet/src/Modulo01.Fundamentos/Ddd/Orders/OrderStatus.cs
namespace JoeDayz.Microservicios.Modulo01.Ddd.Orders;

/// <summary>Ciclo de vida del pedido. Las transiciones validas las controla el agregado.</summary>
public enum OrderStatus
{
    Draft,
    Placed,
    Confirmed,
    Cancelled
}
